ui/src/components/rollouts/Rollout.tsx

```diff
@@ -0,0 +1,129 @@
+import { Menu, Transition } from '@headlessui/react';
+import { EllipsisVerticalIcon } from '@heroicons/react/24/outline';
+import { forwardRef, Fragment, Ref } from 'react';
+import { IRollout, RolloutType } from '~/types/Rollout';
+import { ISegment } from '~/types/Segment';
+import { classNames } from '~/utils/helpers';
+import QuickEditRolloutForm from './QuickEditRolloutForm';
+
+type RolloutProps = {
+  flagKey: string;
+  rollout: IRollout;
+  segments: ISegment[];
+  onQuickEditSuccess: () => void;
+  onEdit: () => void;
+  onDelete: () => void;
+  style?: React.CSSProperties;
+  className?: string;
+  readOnly?: boolean;
+};
+
+const Rollout = forwardRef(
+  (
+    {
+      flagKey,
+      rollout,
+      segments,
+      onQuickEditSuccess,
+      onEdit,
+      onDelete,
+      style,
+      className,
+      readOnly,
+      ...rest
+    }: RolloutProps,
+    ref: Ref<HTMLLIElement>
+  ) => (
+    <li
+      key={rollout.id}
+      ref={ref}
+      style={style}
+      className={`${className} w-full items-center space-y-2 rounded-md border p-2 shadow-md shadow-violet-100 bg-white border-violet-300 hover:shadow-violet-200 sm:flex sm:flex-col lg:px-6 lg:py-2`}
+    >
+      <div className="w-full border-b p-2 bg-white border-gray-200 ">
+        <div className="flex w-full flex-wrap justify-between sm:flex-nowrap">
+          <span
+            key={rollout.id}
+            className={classNames(
+              readOnly ? 'hover:cursor-not-allowed' : 'hover:cursor-move',
+              'ml-2 flex h-4 w-4 justify-start text-gray-400 hover:text-violet-300'
+            )}
+            {...rest}
+          >
+            {rollout.rank}
+          </span>
+          <h3 className="text-sm font-normal leading-6 text-gray-700">
+            {RolloutType[rollout.type as unknown as keyof typeof RolloutType]}{' '}
```

Comment:
I see this kind of stuff and think... what is happening Typescript 😂

Reply:
i think im typescripting wrong.. but idk of another way